src/runtime/utils.ts

```diff
@@ -1,13 +1,26 @@
 export const INTERNAL_PREFIX = "/_frsh";
-export const STATIC_PREFIX = `/static`;
+export const ASSET_CACHE_BUST_KEY = "__frsh_c";
 
 export const IS_BROWSER = typeof document !== "undefined";
 
 /**
- * Return a path "hashed" with the BUILD_ID. of a static file
- * Such path will be served by the server with a Cache-Control of 1 year.
- * @param path a path to a file in the static folder asset. e.g. /style.css
+ * Create a "locked" asset path. This differs from a plain path in that it is
+ * specific to the current version of the application, and as such can be safely
+ * served with a very long cache lifetime (1 year).
  */
 export function asset(path: string) {
-  return `${INTERNAL_PREFIX}${STATIC_PREFIX}/${__FRSH_BUILD_ID}${path}`;
+  if (!path.startsWith("/") || path.startsWith("//")) return path;
+  try {
+    const url = new URL(path, "https://freshassetcache.local");
+    if (
+      url.protocol !== "https:" || url.host !== "freshassetcache.local" ||
+      url.searchParams.has(ASSET_CACHE_BUST_KEY)
+    ) {
+      return path;
+    }
+    url.searchParams.set(ASSET_CACHE_BUST_KEY, __FRSH_BUILD_ID);
+    return url.pathname + url.search + url.hash;
+  } catch {
+    return path;
```

Comment:
maybe add a `console.warn` otherwise it will be hard for anyone to spot that it is not working as expected.